src/arviz_plots/backend/plotly/legend.py

```diff
@@ -36,25 +63,84 @@ def legend(
     None
         The legend is added to the target figure inplace
     """
-    if artist_kwargs is None:
-        artist_kwargs = {}
+    figure = plot_collection.get_viz("figure")
+    if "legend" in plot_collection.viz.children:
+        legend_number = len(plot_collection.viz["legend"].data_vars) + 1
+        legend_id = f"legend{legend_number}"
+    else:
+        legend_number = 1
+        legend_id = "legend"
+    kwargs.setdefault("legend_y", {1: 1, 2: 0, 3: 0.5}[legend_number])
+    kwargs["legend_title_text"] = title
+    legend_kwargs = kwargs.pop(legend_id, {}).copy()
+    kwargs_list = list(kwargs.items())
+    for key, value in kwargs_list:
```

Comment:
Would this be a bit more explicit?
```suggestion
    for key, value in kwargs.copy().items():
```